src/rules/valid-context-access.ts

```diff
@@ -0,0 +1,113 @@
+import { createRule } from "../utils"
+import { extractContextReferences } from "./reference-helpers/svelte-context"
+import type { TSESTree } from "@typescript-eslint/types"
+
+export default createRule("valid-context-access", {
+  meta: {
+    docs: {
+      description:
+        "context functions must be called during component initialization.",
+      category: "Possible Errors",
+      // TODO Switch to recommended in the major version.
+      recommended: false,
+    },
+    schema: [],
+    messages: {
+      unexpected:
+        "Do not call {{function}} except during component initialization.",
+    },
+    type: "problem",
+  },
+  create(context) {
```

Comment:
It probably doesn't work well other then `*.svelte` files, so I think the following guard is needed.

```suggestion
  create(context) {
    if (!context.parserServices.isSvelte) {
      return {}
    }
```

Reply:
I need to add more test but even JS file can use `onMount` so I think we need to check JS files also.

```js
import { setContext, onMount } from "svelte"

const something = () => {
  setContext("answer", 42)
}

onMount(() => something())

```

But at the same time, I found this pattern.
It should be work but now ESLint error occurs.
So I need to handle this.

```js
import { setContext } from "svelte"

const something = () => {
  setContext("answer", 42)
}

const wrapper = (fn) => {
  fn()
}

wrapper(() => something())
```

In addition, I forgot to handle async/await.😇

```js
const something = async () => {
  await Promise.resolve()
  setContext("answer", 42)
}

something()
```

Reply:
I added this logic and I wrote this limitation on the docs.